switchboard/base.py

```diff
@@ -32,12 +32,17 @@ def __init__(self, cache=None, timeout=30):
     def __getitem__(self, key):
         self._populate()
         try:
-            return self._cache[key]
+            value = self._cache[key]
         except KeyError:
             value = self.get_default(key)
             if value is NoValue:
                 raise
-            return value
+        except TypeError:
```

Comment:
Calling dictionary methods (specifically, `__getitem__`) on a `None` object. More broadly, I don't know, which is why I've added logging. Based on what I see in the code, `self._cache` should never be `None` after `self.populate()` is called.